File: selection-sort.php
<?php
$values= array(1,4,10,3,30,14,5);
$size = count($values);

for($i=0;$i<=$size;$i++)
  {
	for($j=$i+1;$j<=$size;$j++)
       {
        if($values[$j] < $values[$i])
        {
            $min = $values[$j];
            $values[$j] = $values[$i];
            $values[$i] = $min;
        }    
       }
    print $values[$i]."\n";
      
  }
   

File: archive/k/kotlin/selectionSort.kt
fun main(args: Array<String>) 
{
    var nums: IntArray
    try
    {
        nums = readLine()!!.split(", ").map{ it.toInt() }.toIntArray()
    }
    catch(e: Exception)
    {
        println("Usage: please provide a list of at least two integers to sort in the format “1, 2, 3, 4, 5”")
        return
    }
    var min:Int
    for(i in 0 until nums.count()-1)
    {
        min = i
        for(j in i+1 until nums.count())
        {
            if(nums[j]<nums[min])
            {
                min = j
            }
            if(min!=i)
            {
                nums[i] = nums[min].also {nums[min] =  nums[i]}
            }
        }
    }
    for(i in 0 until nums.count())
    {
        if (i==nums.count()-1)
        {
            println("${nums[i]}")
            return
        }
        print("${nums[i]}, ")
    }
}